modules/aggregation-layers/src/aggregation-layer-v9/aggregation-layer.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  CompositeLayer,
+  LayerDataSource,
+  LayerContext,
+  UpdateParameters,
+  CompositeLayerProps,
+  Attribute,
+  AttributeManager
+} from '@deck.gl/core';
+import {Aggregator} from './aggregator';
+
+// TODO
+type GPUAggregator = Aggregator & {destroy(): void};
+// TODO
+type CPUAggregator = Aggregator;
+
+export type AggregationLayerProps<DataT> = CompositeLayerProps & {
+  data: LayerDataSource<DataT>;
+};
+
+export default abstract class AggregationLayer<
```

Comment:
Have you thought about how this could work with a `TileLayer`? I'm working on a cluster layer right now which works with a tiled data source, and aggregates whenever the visible data changes (outputting a single `ScatterplotLayer`)